common/network-common/src/test/java/org/apache/spark/network/crypto/TransportCipherTest.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.network.crypto;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.embedded.EmbeddedChannel;
+import org.apache.commons.crypto.cipher.CryptoCipher;
+import org.apache.commons.crypto.stream.CryptoInputStream;
+import org.apache.commons.crypto.stream.CryptoOutputStream;
+import org.apache.commons.crypto.stream.input.ChannelInput;
+import org.apache.spark.network.util.MapConfigProvider;
+import org.apache.spark.network.util.TransportConf;
+import org.hamcrest.CoreMatchers;
+import org.junit.Test;
+
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.channels.WritableByteChannel;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class TransportCipherTest {
+
+  @Test
+  public void testBufferNotLeaksOnInternalError() throws IOException {
+    String algorithm = "TestAlgorithm";
+    TransportConf conf = new TransportConf("Test", MapConfigProvider.EMPTY);
+    TransportCipher cipher = new TransportCipher(conf.cryptoConf(), conf.cipherTransformation(),
+            new SecretKeySpec(new byte[256], algorithm), new byte[0], new byte[0]) {
+
+      @Override
+      CryptoOutputStream createOutputStream(WritableByteChannel ch) {
+        return null;
+      }
+
+      @Override
+      CryptoInputStream createInputStream(ReadableByteChannel ch) throws IOException {
+        CryptoCipher cipher = mock(CryptoCipher.class);
+        when(cipher.getBlockSize()).thenReturn(8);
+        when(cipher.getAlgorithm()).thenReturn(algorithm);
+
+        return new CryptoInputStream(new ChannelInput(ch), cipher, 1024,
+                null, new IvParameterSpec(new byte[0])) {
+          @Override
+          public int read(ByteBuffer dst) {
+            // Simulate throwing InternalError
+            throw new InternalError();
```

Comment:
@normanmaurer . `dev/lint-java` complains at this line and line 76.
```
[ERROR] src/test/java/org/apache/spark/network/crypto/TransportCipherTest.java:[70] (regexp) RegexpSinglelineJava: Avoid throwing error in application code.
[ERROR] src/test/java/org/apache/spark/network/crypto/TransportCipherTest.java:[76] (regexp) RegexpSinglelineJava: Avoid throwing error in application code.
```

Reply:
I know but I want to throw this as the test actually test exactly for this scenario ;)

Reply:
Can we use `*Exception` instead in this test code?

Reply:
The commons-crypto library throws `InternalError` so to test that behavior you have to use that error.

Reply:
Found somewhere that you can use `// CHECKSTYLE:OFF` and `// CHECKSTYLE:ON` comments to disable checkstyle for some code.

Reply:
Got it. +1 for @vanzin 's advice.

Reply:
Just pushed this change... lets see if it works :)